libevmasm/Assembly.cpp

```diff
@@ -1344,12 +1344,12 @@ std::map<uint16_t, uint16_t> Assembly::findReferencedContainers() const
 				referencedSubcontainersIds.insert(containerId);
 			}
 
-	std::map<uint16_t, uint16_t> replacements;
+	std::map<ContainerID, ContainerID> replacements;
 	uint8_t nUnreferenced = 0;
-	for (uint8_t i = 0; i < static_cast<uint16_t>(m_subs.size()); ++i)
+	for (uint8_t i = 0; i < static_cast<ContainerID>(m_subs.size()); ++i)
```

Comment:
`i` will overflow and the loop will never stop when there are 256 sections. You need a bigger type to iterate over sections:

```suggestion
	for (size_t i = 0; i < m_subs.size(); ++i)
```

Would be best if we could have this corner case covered with a test. Maybe one with empty subobjects to keep it compact?

Reply:
Sure. Shit I assumed for some reason that `0x100 == 100` 👯

Reply:
This really makes me question whether not investing more time into tests at this stage is really a wise decision. Ultimately things like this should be detected by tests. We're rushing towards semantic tests, but some cases like this one definitely won't be caught by them. And these smaller PRs are the only place to consider corner cases in detail. Once it's all in, no one will have time to go back through the whole backend and figure out where extra tests are needed.

Reply:
Adding tests too.

Reply:
This reminds me also to add proper error to `AsmAnalysis` covering the case when we have more then 256 subcontainers.

Reply:
Done